app/src/androidTest/java/com/github/se/project/ui/End2End/TutorEndToEnd.kt

```diff
@@ -0,0 +1,124 @@
+  // Section.GM, academicLevel = AcademicLevel.BA1, languages = listOf(), subjects = listOf(),
+  // schedule = listOf())
+  // var currentProfile: Profile? = null
+
+  private val mockUid = "mockUid"
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    context = mock(Context::class.java)
+    whenever(mockProfileRepository.addProfile(any(), any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    whenever(mockProfileRepository.getNewUid()).thenReturn("mockUid")
+  }
+
+  @Test
+  fun TutorEndToEndTest() {
+    composeTestRule.setContent {
+      PocketTutorApp(true, viewModel(), mockProfileViewModel, mockLessonViewModel)
+    }
+    // Sign In Screen
+    composeTestRule.onNodeWithTag("logo").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("loginButton").performClick()
+    composeTestRule.onNodeWithTag("firstNameField").assertIsDisplayed()
+
+    // Create Profile Screen
+    composeTestRule.onNodeWithTag("firstNameField").performTextInput("John")
+    composeTestRule.onNodeWithTag("lastNameField").performTextInput("Doe")
+    composeTestRule.onNodeWithTag("phoneNumberField").performTextInput("0213456789")
+    composeTestRule.onNodeWithTag("roleButtonTutor").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdown").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdownItem-SC").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdown").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdownItem-BA3").performClick()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+    assertEquals(Role.TUTOR, mockProfileViewModel.currentProfile.value?.role)
+
+    // Create Tutor Profile Screen
+    composeTestRule.onNodeWithTag("welcomeText").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("checkbox_FRENCH").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("dropdownANALYSIS").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("priceSlider").performGesture { swipeRight() }
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+
+    // Create Tutor Schedule Screen
+    composeTestRule
+        .onNodeWithTag("welcomeText")
+        .assertTextEquals("John, show us your availabilities")
+    composeTestRule.onNodeWithTag("Slot_0_0").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_3").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_2").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_6").performClick()
+    composeTestRule.onNodeWithTag("FindStudentButton").performClick()
+
+    // Home Screen
+    composeTestRule.onNodeWithContentDescription("Profile Icon").performClick()
```

Comment:
Should probably indicate with a comment that you're moving to the profile info screen